### DISCUSSION TITLE
02_classsification_in_pytorch 9:54:31

### DISCUSSION THREAD

**Discussion by crice6719** [mrdbourke/pytorch-deep-learning#295] [Q&A]
When trying to make predictions and print untrained preds, im getting a mat1 and ma2 must have same dtype error. I have created all the tensors of the float dtype so im not sure what might be causing this error. Any help would be greatly appreciated. 

[Link ](https://github.com/crice6719/Lean/blob/72842882434095a2606bbd767f85a598ff0e21ca/classification_in_pytorch.ipynb)to the code

**Comment by tonyhaya** [comment]
Try to tranpose one of matrix `torch.transpose()`, https://pytorch.org/docs/stable/generated/torch.transpose.html

**Comment by mrdbourke** [comment]
Hi @crice6719,

The default datatype in PyTorch for models/data is `torch.float32`, however, I see you're using `torch.float64`.

Have you tried using `torch.float32`?

That may help (as I think the error comes from your model rather than your data).

In essence, it looks like your data is `torch.float64` but your model will be `torch.float32` (by default).
